.github/actions/archive-surefire-reports/action.yml

```diff
@@ -37,7 +37,7 @@ runs:
     - id: upload-surefire-linux
       name: Upload Surefire reports
       if: (!cancelled() && contains(fromJSON(inputs.release-branches), github.ref) && contains(fromJSON('["push", "workflow_dispatch"]'), github.event_name))
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
```

Comment:
@ahus1 this one is used for getting test date for RHBK release, is the v4 stable? thanks

Reply:
TL;DR: We can stay on the safe side an downgrade this to v3 as well.

I learned that you shouldn't mix the upload/download artifact from different versions. On the external API on how to download files, those should be ok from what I see, still I saw a problem with the flaky tests detection. Some say that your access token needs to have the "action" permission set. Unfortunately I don't have access to Keycloak's GitHub action deployment logs, so I postponed that.

Reply:
Based on the v4 limit it's not usable, because there may be more then 10 artifacts per job.

Reply:
@ahus1 it seems that it may work, because it is limited per artifact: https://github.com/actions/upload-artifact/blob/cf8714cfeaba5687a442b9bcb85b29e23f468dfa/README.md#number-of-artifacts, not per number of files within artifact, checked release branch 22.0 -> https://github.com/keycloak/keycloak/actions/runs/7247061912, let's go with v4

Reply:
Looking at the example of https://github.com/keycloak/keycloak/actions/runs/7247061912 it is one archive per job which for the tests, which is IMHO ok. 

I also tested downloading the archive uploaded by v4 for the maven artifacts with the GH CLI, and it works as expected without problems. 

```
gh api https://api.github.com/repos/keycloak/keycloak/actions/artifacts/1127338868/zip > out.tzts
```

Reply:
I've also checked the Jenkins GH token permissions -> it has action permission right now, but is generated from my account -> default Jenkins gh token doesn't have such privileges enabled.